jenkins_pipelines/uyuni_prs/Jenkinsfile_prs_susemanager_unittest

```diff
@@ -51,21 +41,22 @@ pipeline {
         stage('Check Pull Request') {
             steps {
                  echo 'Check if a PR need a test'
-                 sh "${check} | grep \"TESTREQUIRED=true\" "
+                 sh "${check} 2>&1 | tee gitarro_check.log"
                }
          }
         // THIS JOB RUN ONLY IF THE 1ST IS SUCCESSEFULL. ( so it is triggered by 1st)
         stage('Run tests') {
+            when {
+                expression { return readFile('gitarro_check.log').contains('TESTREQUIRED=true') }
+            }
             steps {
                   echo 'Run tests'
-                   sh "$runtest}"
-                  }
+                  sh "${runtest}"
+                  junit "**/susemanager/src/reports/tests.xml"
```

Comment:
I agree with @MalloZup comment. IIRC, if a sh step fails, the whole build fails.

However having the junit always working on post is ugly, since it produces a stack trace, so if the build fails when the sh fails, I guess  will need to trap the return code and return it after junit.

Reply:
if the runtest fails, we will not have junit so we want to skip the collection of it.
if the runtest pass, we want to continue collecting using junit step.
So, I can put I try catch here just to avoid a failure in the job if the runtest fails, but we don't want to just fail the job if the runtest fails?

Reply:
if the test fail you need to collect results for having tests results

Reply:
e.g line 55 will never called when 54 fails. You need to store the results then fail

Reply:
if the bash process runtest fails, it will not produce any test results right?
if the runtest has some test failing, it will anywait exit with 0, so it will not fail as a bash process, then it will continue with next step "junit".

Reply:
afaik if it fail it can produce tests. ( but yop if your assumption is right this is right), i would checkout that point